tensorflow/python/kernel_tests/unique_op_test.py

```diff
@@ -152,6 +152,24 @@ def testShapeInferenceV2(self):
     _, idx = gen_array_ops.unique_v2(x, axis=axis_t)
     self.assertEqual(idx.shape.as_list(), [None])
 
+  def testEmpty(self):
+    x = np.random.randint(2, size=0)
+    y, idx = array_ops.unique(x)
+    tf_y, tf_idx = self.evaluate([y, idx])
+
+    self.assertEqual(len(x), len(tf_idx))
+    self.assertEqual(len(tf_y), len(np.unique(x)))
+
+  def testOrderedByAppearance(self):
+    x = np.array([3, 5, 3, 4, 1, 4, 9, 8, 6, 3, 5, 7, 8, 8, 4, 6, 4, 2, 5, 6])
+    true_y = np.array([3, 5, 4, 1, 9, 8, 6, 7, 2])
+    true_idx = np.array(
+        [0, 1, 0, 2, 3, 2, 4, 5, 6, 0, 1, 7, 5, 5, 2, 6, 2, 8, 1, 6])
+    y, idx = array_ops.unique(x)
+    tf_y, tf_idx = self.evaluate([y, idx])
+    self.assertAllEqual(tf_y, true_y)
+    self.assertAllEqual(tf_idx, true_idx)
```

Comment:
I suspect these are exercising the existing CPU kernel for `Unique` that works on integers.  If that's true, can you please use floats?

Reply:
The type registrations are identical for CPU and GPU, and the tests are running on the GPU for me.